### PR TITLE
NTC Skild/Takbok choice

### PR DESCRIPTION
**Pull request by projectkepler-ru**
## About The Pull Request
gives  you the option to pick between skild and takbok as NTC. Should work fine
This also work as a frame work that can be used by anyone in the future to add gun choice for NTC, as  I don't believe one gun can covers the need of everyone

## How This Contributes To The Skyrat Roleplay Experience
Because what if my NTC is a revolver person and not a handgun person? I might just prefer one over the other. It's a small differences but I've actually ask myself and everyone alot in game and off game during playing as hos/captain "Am I a revolver or a handgun person" and while people often replied humerously like "get a grenade launcher" or "use a lush wish whip" I really want people to be able to make that choice and have that same question I did. So, revolver or handgun?

## Proof of Testing
image
image
compiled and ran just fine on my end during testing so it should be fine

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog

:cl:
add: a gun beacon option for the NTC to pick between two of his magnum pistol choices, fully modular and can be messed with by anyone in the future if more guns are added.
/:cl:

### CODE REVIEW

**Comment by Plum-but-gayer** [comment]
This is a fun little flavor deal, I'd love to see more stuff like this giving people round start gear choices (like how miners get to pick their first item, etc.) Bring back sec gun tickets :)

**Comment by projectkepler-ru** [comment]
@Plum-but-gayer 
the thing I wrote on why it's good for RP isn't some weird shitpost either, I genuinely didn't know if I preferred revolver or a handgun so now I'm forcing you all to suffer the same philosophy

and while I debated giving the captain this too- it became apparent that it doesn't really matter
the captain is allowed to spend the budget however he pleases and buy whatever he wants; meaning he can just buy the revolver or pistol if he so choose, where asto the consultant doesn't really have that much money on hand and trying to save up to buy the revolver and ammobox later is quite a hassle because by the time you made that decision you likely wanted to buy something else, while you could also just ask people for money or ask the captain to buy the gun for you, that kinda defeats the point, so this is the best outcome.

in contrast to bubberstation where you aren't allowed to use command budget to buy stuff for yourself. So I kept that in mind.

**Comment by desertmote** [comment]
Not a fan. I play this role too and can buy my weapon of choice 10 minutes in without bounties. (wespe)
gun beacons returning seems like a slippery slope to me. You mention bubber in your comment, why not PR this there?

~~NTC is often supposed to be a non-combatant paper pusher. It makes sense their gun is an awful to use gun from Sol. Giving them an NRI/CIN(forgot if they were separate) gun that is actually pretty meta mechanically doesnt click right.~~

**Comment by Paxilmaniac** [comment]
> Giving them an NRI/CIN(forgot if they were separate) gun that is actually pretty meta mechanically doesnt click right.

The Takbok is still a Sol gun, its just a revolver version of the pistol they spawn with

**Comment by desertmote** [comment]
> > Giving them an NRI/CIN(forgot if they were separate) gun that is actually pretty meta mechanically doesnt click right.
> 
> The Takbok is still a Sol gun, its just a revolver version of the pistol they spawn with

My bad, I confused it for the shotgun shell revolver, disregard that part of my comment.

**Comment by projectkepler-ru** [comment]
> Not a fan. I play this role too and can buy my weapon of choice 10 minutes in without bounties. (wespe) gun beacons returning seems like a slippery slope to me. You mention bubber in your comment, why not PR this there?
> 
> NTC is often supposed to be a non-combatant paper pusher. It makes sense their gun is an awful to use gun from Sol. Giving them an NRI/CIN(forgot if they were separate) gun that is actually pretty meta mechanically doesnt click right.

I mean that's your choice if you want to use wespe I'm just giving them the pick between the two magnum pistol options

and I mentioned bubber  quite literally with the reasons stated in it

I'm not doing that there because if they want to add something like this they'd do it there. I don't want to get too involved


Also I'd greatly appreciated if you actually downloaded my PR and checked it out locally OR looked at this part before saying I was adding an NRI gun
image

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm [hunk 1]
```diff
@@ -152,3 +152,35 @@
...
+	new /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/skild(src)
+	new /obj/item/storage/fancy/cigarettes/cigars/havana(src)
+
+/obj/item/storage/box/ntc/revolver
+	name = "Takbok Set."
+	desc = "Contains your .585 revolver and a cigar case."
+
+/obj/item/storage/box/ntc/revolver/PopulateContents()
+	new /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/takbok(src)
+	new /obj/item/storage/fancy/cigarettes/cigars/havana(src)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]: Why not just spawn in the gun boxes instead? No need to do things like this, just spawn the guncase.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm (hunk 1 above)
Comment (projectkepler-ru) [reply modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]: Oh sure

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm [hunk 2]
```diff
@@ -152,3 +152,18 @@
...
 	new /obj/item/storage/bag/garment/nanotrasen_consultant(src)
+
+//Choice Beacon, I hope in the future they're going to be given proper unique gun but this will do.
+
+
+/obj/item/choice_beacon/ntc //I promise you, this will solve everything
+	name = "gunset beacon" //Haven't you done this before?
+	desc = "A single use beacon to deliver a gunset of your choice. Please only call this in your office" // Done what?
+	company_source = "Trappiste Fabriek Company" // You know, tried adding gun beacon prior
+	company_message = span_bold("Supply Pod incoming please standby") //Did I?
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]:
```suggestion
	company_message = span_bold("Supply Pod incoming, please stand by") //Did I?
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm [hunk 3]
```diff
@@ -152,3 +152,19 @@
...
 	new /obj/item/storage/bag/garment/nanotrasen_consultant(src)
+
+//Choice Beacon, I hope in the future they're going to be given proper unique gun but this will do.
+
+
+/obj/item/choice_beacon/ntc //I promise you, this will solve everything
+	name = "gunset beacon" //Haven't you done this before?
+	desc = "A single use beacon to deliver a gunset of your choice. Please only call this in your office"
+	company_source = "Trappiste Fabriek Company"
+	company_message = span_bold("Supply Pod incoming please standby")
```
Comment (projectkepler-ru) [inline modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]:
```suggestion
	company_message = span_bold("Supply Pod incoming please stand by")
```
Nvm should've changed it to that

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm [hunk 4]
```diff
@@ -152,3 +152,19 @@
...
+/obj/item/choice_beacon/ntc //I promise you, this will solve everything
+	name = "gunset beacon" //Haven't you done this before?
+	desc = "A single use beacon to deliver a gunset of your choice. Please only call this in your office"
+	company_source = "Trappiste Fabriek Company"
+	company_message = span_bold("Supply Pod incoming please stand by")
+
+/obj/item/choice_beacon/ntc/generate_display_names()
+	var/static/list/selectable_gun_types //What's wrong with it? if I did.
+	if(!selectable_gun_types) //Well I think it was a tad overpowered?
+		selectable_gun_types = list("Takbok" = /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/takbok, "Skild" = /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/skild) //But shouldn't the HoS be armed with the best gear at all time
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]:
```suggestion
	var/static/list/selectable_gun_types = list(
		"Takbok" = /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/takbok,
		"Skild" = /obj/item/storage/toolbox/guncase/skyrat/pistol/trappiste_small_case/skild,
	)
```

File: modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm [hunk 5]
```diff
@@ -152,3 +152,19 @@
 	new /obj/item/storage/photo_album/personal(src)
 	new /obj/item/bedsheet/centcom(src)
 	new /obj/item/storage/bag/garment/nanotrasen_consultant(src)
+
+//Choice Beacon, I hope in the future they're going to be given proper unique gun but this will do.
+
+
+/obj/item/choice_beacon/ntc //I promise you, this will solve everything
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/nanotrasen_rep/code/nanotrasen_consultant.dm]: Also, maybe chill out with the weird commentary on the sides of this, it just adds more commentary that doesn't serve any purpose.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by GoldenAlpharex — APPROVED** [review]
Looks good now!

**Pull request merged**